sdk/lib/rtl/process.c

```diff
@@ -493,3 +493,19 @@ RtlGetCurrentProcessorNumber(VOID)
     /* Forward to kernel */
     return NtGetCurrentProcessorNumber();
 }
+
+ULONG
+NTAPI
+RtlGetAppcompatVersion(VOID)
+{
+    /* Get the current PEB */
+    PPEB Peb = RtlGetCurrentPeb();
+    if (Peb == NULL)
+    {
+        /* Default to Server 2003 */
+        return _WIN32_WINNT_WS03;
+    }
+
+    /* Calculate OS version from PEB fields */
+    return (Peb->OSMajorVersion << 8) | Peb->OSMinorVersion;
```

Comment:
I am assuming this is supposed to be substituted by the selected appcompat version later?

Reply:
Well I plan to actually patch the PEB.

Reply:
Can we make it not look like it's an "official" RTL function?

Reply:
Renamed it to RtlRosGetAppcompatVersion(). Everyone knows: if there is a "Ros" in there, it's a hack :)